### PR TITLE
Clean up array_unique & clarify cast behaviour

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: blackbox/docs/sql/scalar.txt [hunk 1]
```diff
@@ -1206,7 +1206,7 @@ The ``array_unique`` function merges two arrays into one array with unique eleme
 
 ::
 
-    cr> select array_unique([1,2,3],[3,4,5,6]) from sys.cluster;
+    cr> select array_unique([1, 2, 3], [3, 4, 5, 6]) from sys.cluster;
```
Comment (matriv) [inline blackbox/docs/sql/scalar.txt]: I would also add duplicate values inside the arrays to show the uniquness

File: sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java [hunk 2]
```diff
@@ -22,201 +22,101 @@
...
-        assertEval(
-            new Object[]{1},
-            Literal.of(new Object[]{1}, arrayOfIntegerType),
-            Literal.of(new Object[]{1L}, arrayOfLongType));
+        assertEvaluate("array_unique([10, 20], [10.1, 20.0])", new Long[] {10L, 20L });
     }
 
     @Test
     public void testConvertNonNumericStringToNumber() throws Exception {
         expectedException.expect(NumberFormatException.class);
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java]: Maybe catch and convert to another exception with more descriptive message

File: blackbox/docs/sql/scalar.txt (hunk 1 above)
Comment (matriv) [inline blackbox/docs/sql/scalar.txt]: I would also add duplicate values inside the arrays to show the uniquness

File: sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java [hunk 3]
```diff
@@ -22,201 +22,101 @@
...
-
-    @Test
     public void testEmptyArrays() throws Exception {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage("One of the arguments of the array_unique function can be of undefined inner type, but not both");
-        assertEval(new Object[]{},
-            Literal.of(new Object[]{}, arrayOfUndefinedType),
-            Literal.of(new Object[]{}, arrayOfUndefinedType));
+        expectedException.expect(UnsupportedOperationException.class);
+        expectedException.expectMessage("One of the arguments of the array_unique function can " +
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java]: Maybe rephrase: No more than on arguments...

File: blackbox/docs/sql/scalar.txt (hunk 1 above)
Comment (mfussenegger) [inline blackbox/docs/sql/scalar.txt]: 3 is a duplicate?

File: sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java (hunk 2 above)
Comment (mfussenegger) [inline sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java]: I'd prefer to not have any behaviour changes within this PR

File: blackbox/docs/sql/scalar.txt (hunk 1 above)
Comment (matriv) [inline blackbox/docs/sql/scalar.txt]: mmm I mean inside the subarray: like [3, 4, 5, 3, 6, 4] or am I missing something?

File: sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java (hunk 2 above)
Comment (matriv) [inline sql/src/test/java/io/crate/operation/scalar/ArrayUniqueFunctionTest.java]: ok
